system-parachains/asset-hubs/asset-hub-polkadot/src/ah_migration/call_filter.rs

```diff
@@ -0,0 +1,210 @@
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Call filters for Asset Hub during the Asset Hub Migration.
+
+use crate::*;
+
+/// Contains all calls that are enabled before the migration.
+pub struct CallsEnabledBeforeMigration;
+impl Contains<<Runtime as frame_system::Config>::RuntimeCall> for CallsEnabledBeforeMigration {
+	fn contains(call: &<Runtime as frame_system::Config>::RuntimeCall) -> bool {
+		let (before, _, _) = call_allowed_status(call);
+		if !before {
+			log::warn!("Call bounced by the filter before the migration: {:?}", call);
+		}
+		before
+	}
+}
+
+/// Contains all calls that are enabled during the migration.
+pub struct CallsEnabledDuringMigration;
+impl Contains<<Runtime as frame_system::Config>::RuntimeCall> for CallsEnabledDuringMigration {
+	fn contains(call: &<Runtime as frame_system::Config>::RuntimeCall) -> bool {
+		let (_before, during, _after) = call_allowed_status(call);
+		if !during {
+			log::warn!("Call bounced by the filter during the migration: {:?}", call);
+		}
+		during
+	}
+}
+
+/// Contains all calls that are enabled after the migration.
+pub struct CallsEnabledAfterMigration;
+impl Contains<<Runtime as frame_system::Config>::RuntimeCall> for CallsEnabledAfterMigration {
+	fn contains(call: &<Runtime as frame_system::Config>::RuntimeCall) -> bool {
+		let (_before, _during, after) = call_allowed_status(call);
+		if !after {
+			log::warn!("Call bounced by the filter after the migration: {:?}", call);
+		}
+		after
+	}
+}
+
+/// Return whether a call should be enabled before, during and/or after the migration.
+///
+/// Time line of the migration looks like this:
+///
+/// --------|-----------|--------->
+///       Start        End
+///
+/// We now define 3 periods:
+///
+/// 1. Before the migration: [0, Start)
+/// 2. During the migration: [Start, End]
+/// 3. After the migration: (End, ∞)
+///
+/// Visually:
+///
+/// ```text
+/// |--1---|
+///         |-----2-----|
+///                      |---3---->
+/// --------|-----------|--------->
+///       Start        End
+/// ```
+///
+/// This call returns a 3-tuple to indicate whether a call is enabled during these periods.
+pub fn call_allowed_status(
+	call: &<Runtime as frame_system::Config>::RuntimeCall,
+) -> (bool, bool, bool) {
+	use RuntimeCall::*;
+	const ON: bool = true;
+	const OFF: bool = false;
+	let before_migration = call_allowed_before_migration(call);
+
+	let during_migration = match call {
+		AhMigrator(..) => ON, // required for the migration, only permissioned calls
+		AhOps(..) => OFF,     // Not needed during the migration
+		AssetConversion(..) => ON, // no reason to disable it, just convenience
+		AssetRate(..) => OFF,
+		Assets(..) => ON,   // no reason to disable it, just convenience
+		Balances(..) => ON, // no reason to disable it, just convenience
+		Bounties(..) => OFF,
+		ChildBounties(..) => OFF,
+		Claims(..) => OFF,
+		CollatorSelection(..) => ON, // Why?
```

Comment:
Could this be clarified? Most others say `// no reason to disable it, just convenience` - why not the same here?